pages/company/visualizations/company_associated_activity.py

```diff
@@ -0,0 +1,215 @@
+                                    ),
+                                    width="auto",
+                                ),
+                                dbc.Col(
+                                    dbc.Button(
+                                        "About Graph",
+                                        id=f"{PAGE}-popover-target-{VIZ_ID}",
+                                        color="secondary",
+                                        size="sm",
+                                    ),
+                                    width="auto",
+                                    style={"paddingTop": ".5em"},
+                                ),
+                            ],
+                            align="center",
+                            justify="between",
+                        ),
+                    ]
+                ),
+            ]
+        )
+    ],
+)
+
+# callback for graph info popover
+@callback(
+    Output(f"{PAGE}-popover-{VIZ_ID}", "is_open"),
+    [Input(f"{PAGE}-popover-target-{VIZ_ID}", "n_clicks")],
+    [State(f"{PAGE}-popover-{VIZ_ID}", "is_open")],
+)
+def toggle_popover(n, is_open):
+    if n:
+        return not is_open
+    return is_open
+
+
+# callback for Company Affiliation by Github Account Info graph
+@callback(
+    Output(VIZ_ID, "figure"),
+    [
+        Input("repo-choices", "data"),
+        Input(f"{PAGE}-{paramter_1}-{VIZ_ID}", "value"),
+        Input(f"{PAGE}-date-picker-range-{VIZ_ID}", "start_date"),
+        Input(f"{PAGE}-date-picker-range-{VIZ_ID}", "end_date"),
+    ],
+    background=True,
+)
+def compay_associated_activity_graph(repolist, num, start_date, end_date):
+
+    # wait for data to asynchronously download and become available.
+    cache = cm()
+    df = cache.grabm(func=cmq, repos=repolist)
+    while df is None:
+        time.sleep(1.0)
+        df = cache.grabm(func=cmq, repos=repolist)
+
+    start = time.perf_counter()
+    logging.debug(f"{VIZ_ID}- START")
+
+    # test if there is data
+    if df.empty:
+        logging.debug(f"{VIZ_ID} - NO DATA AVAILABLE")
+        return nodata_graph
+
+    # function for all data pre processing, COULD HAVE ADDITIONAL INPUTS AND OUTPUTS
+    df = process_data(df, num, start_date, end_date)
+
+    fig = create_figure(df)
+
+    logging.debug(f"{VIZ_ID} - END - {time.perf_counter() - start}")
+    return fig
+
+
+def process_data(df: pd.DataFrame, num, start_date, end_date):
+
+    # convert to datetime objects rather than strings
+    df["created"] = pd.to_datetime(df["created"], utc=True)
+
+    # order values chronologically by COLUMN_TO_SORT_BY date
+    df = df.sort_values(by="created", axis=0, ascending=True)
+
+    # filter values based on date picker
+    if start_date is not None:
+        df = df[df.created >= start_date]
+    if end_date is not None:
+        df = df[df.created <= end_date]
+
+    # creates list of emails for each contribution and flattens list result
```

Comment:
Add documentation in docstring or comment here:

"Each contribution is associated with a contributor. That contributor can be associated with more than one different email. Hence each contribution is associated with all of the emails that a contributor has historically used.

We don't always know which email (and therefore which organization) a contributor is affiliated with at contribution time, so we choose to count all of their possible affiliations via their email list. 

e.g. if "Jane Doe" is associated with "gmail.com" and "yahoo.com" and they have 5 contributions, "gmail.com" and "yahoo.com" would be counted 5 times each.

We assume that relatively few people will have many emails. We acknowledge that this will almost always contribute to an overcount but will never undercount."

In the future, we should add the option to (1/n) the counts where 'n' is the number of emails 
a contributor has.